PocketKit/Sources/PocketKit/MyList/ItemsList/PocketItemViewModel.swift

```diff
@@ -7,78 +7,89 @@ import Analytics
 import Sync
 import SharedPocketKit
 
+/// View model used for List Item to handle actions for a specific item
 class PocketItemViewModel: ObservableObject {
     private let tracker: Tracker
     private let source: Source
+    private let index: Int
+    private let scope: SearchScope
+
     let item: PocketItem
-    let index: Int
 
     @Published
     var isFavorite: Bool
 
     @Published
     var presentShareSheet: Bool = false
 
-    init(item: PocketItem, index: Int, source: Source, tracker: Tracker) {
+    /// Determines whether an item should show Archive or Move To Saves
+    var isArchived: Bool {
+        item.isArchived
+    }
+
+    /// Retrieves view model to present Add Tags view
+    var tagsViewModel: PocketAddTagsViewModel? {
+        guard let savedItem = fetchSavedItem() else {
+            Log.capture(message: "PocketAddTagsViewModel not returned")
+            return nil
+        }
+
+        let addTagsViewModel = PocketAddTagsViewModel(
+            item: savedItem,
+            source: source,
+            saveAction: {}
+        )
+        tracker.track(event: Events.Search.showAddTags(itemUrl: savedItem.url, positionInList: index, scope: scope))
+        return addTagsViewModel
+    }
+
+    init(item: PocketItem, index: Int, source: Source, tracker: Tracker, scope: SearchScope) {
         self.item = item
         self.index = index
         self.source = source
         self.tracker = tracker
+        self.scope = scope
         self.isFavorite = item.isFavorite
     }
 
-    func favoriteAction(index: Int, scope: SearchScope) -> ItemAction {
+    /// Triggers action to favorite or unfavorite an item in a list
+    func favoriteAction() -> ItemAction {
         if isFavorite {
             return .unfavorite { [weak self] _ in
-                self?._unfavorite(index: index, scope: scope)
+                guard let self else { return }
```

Comment:
@cyndichin should we log when self is nil especially since it got you and I the other day.

Reply:
will update!